app/kumactl/cmd/apply/apply.go

```diff
@@ -153,21 +160,29 @@ $ kumactl apply -f https://example.com/resource.yaml
 	return cmd
 }
 
-func upsert(ctx context.Context, typeRegistry registry.TypeRegistry, rs store.ResourceStore, res model.Resource) error {
+func upsert(ctx context.Context, typeRegistry registry.TypeRegistry, rs store.ResourceStore, res model.Resource) ([]string, error) {
 	newRes, err := typeRegistry.NewObject(res.Descriptor().Name)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
+	var warnings []string
+	warnContext := context.WithValue(ctx, remote.WarningsCallback, func(s []string) {
+		warnings = s
+	})
+
 	meta := res.GetMeta()
 	if err := rs.Get(ctx, newRes, store.GetByKey(meta.GetName(), meta.GetMesh())); err != nil {
 		if store.IsResourceNotFound(err) {
-			return rs.Create(ctx, res, store.CreateByKey(meta.GetName(), meta.GetMesh()), store.CreateWithLabels(meta.GetLabels()))
+			cerr := rs.Create(warnContext, res, store.CreateByKey(meta.GetName(), meta.GetMesh()), store.CreateWithLabels(meta.GetLabels()))
+			return warnings, cerr
 		} else {
-			return err
+			return nil, err
 		}
 	}
 	if err := newRes.SetSpec(res.GetSpec()); err != nil {
-		return err
+		return nil, err
 	}
-	return rs.Update(ctx, newRes, store.UpdateWithLabels(meta.GetLabels()))
+	uerr := rs.Update(warnContext, newRes, store.UpdateWithLabels(meta.GetLabels()))
+	return warnings, uerr
```

Comment:
we really need to introduce proper client. It's quite a big workaround. Ofc not a subject of this PR

Reply:
Yeah we need to finish the openAPI spec and just autogen most of the client!